### PR TITLE
Update plotly.js version to 2.29.0 + add Plotly.py docs examples

### PR DESCRIPTION
**Pull request by LiamConnors**
- Update Plotly.js version to 2.29.0

Add docs:
- Examples of `align` with Sankey diagrams
- Add details on using multiple WebGL contexts
- Add changes to using stamen tiles
- Examples of rounded bars
- Example of setting `hovercolor` on sankey links
- Example of `autotickangles`

### Documentation PR

- [x] I've [seen the `doc/README.md` file](https://github.com/plotly/plotly.py/blob/master/doc/README.md)
- [x] This change runs in the current version of Plotly on PyPI and targets the `doc-prod` branch OR it targets the `master` branch
- [x] If this PR modifies the first example in a page or adds a new one, it is a `px` example if at all possible
- [x] Every new/modified example has a descriptive title and motivating sentence or paragraph
- [x] Every new/modified example is independently runnable
- [x] Every new/modified example is optimized for short line count	and focuses on the Plotly/visualization-related aspects of the example rather than the computation required to produce the data being visualized
- [x] Meaningful/relatable datasets are used for all new examples instead of randomly-generated data where possible
- [x] The random seed is set if using randomly-generated data in new/modified examples
- [x] New/modified remote datasets are loaded from https://plotly.github.io/datasets and added to https://github.com/plotly/datasets
- [x] Large computations are avoided in the new/modified examples in favour of loading remote datasets that represent the output of such computations
- [x] Imports are `plotly.graph_objects as go` / `plotly.express as px` / `plotly.io as pio`
- [x] Data frames are always called `df`
- [x] `fig = <something>` call is high up in each new/modified example (either `px.<something>` or `make_subplots` or `go.Figure`)
- [x] Liberal use is made of `fig.add_*` and `fig.update_*` rather than `go.Figure(data=..., layout=...)` in every new/modified example
- [x] Specific adders and updaters like `fig.add_shape` and `fig.update_xaxes` are used instead of big `fig.update_layout` calls in every new/modified example
- [x] `fig.show()` is at the end of each new/modified example
- [x] `plotly.plot()` and `plotly.iplot()` are not used in any new/modified example
- [x] Hex codes for colors are not used in any new/modified example in favour of [these nice ones](https://github.com/plotly/plotly.py/issues/2192)

## Code PR

- [ ] I have read through the [contributing notes](https://github.com/plotly/plotly.py/blob/master/contributing.md) and understand the structure of the package. In particular, if my PR modifies code of `plotly.graph_objects`, my modifications concern the `codegen` files and not generated files.
- [ ] I have added tests (if submitting a new feature or correcting a bug) or
  modified existing tests.
- [ ] For a new feature, I have added documentation examples in an existing or
  new tutorial notebook (please see the doc checklist as well).
- [ ] I have added a CHANGELOG entry if fixing/changing/adding anything substantial.
- [ ] For a new feature or a change in behaviour, I have updated the relevant docstrings in the code to describe the feature or behaviour (please see the doc checklist as well).

-->

### CODE REVIEW

File: doc/python/webgl-vs-svg.md [hunk 1]
```diff
@@ -58,6 +59,15 @@ In addition to the above limitations, the WebGL-powered version of certain SVG-p
...
 * Axis range heuristics may differ
 
+### Multiple WebGL Contexts
+
+*New in 5.19*
+
+If you encounter WebGL context limits when rendering many WebGL-based figures on one page, you can use [Virtual WebGL](https://github.com/greggman/virtual-webgl).
+
+To use it, in the environment where your Plotly figures are being rendered, load the Virtual WebGL script, "https://unpkg.com/virtual-webgl@1.0.6/src/virtual-webgl.js", for example, using a `<script>` tag.
+
```
Comment (archmoj) [inline doc/python/webgl-vs-svg.md:69]: Would you provide an example of using the script?

Comment (alexcjohnson) [reply doc/python/webgl-vs-svg.md:69]: Most importantly, how do you include this script in Jupyter (lab or classic notebook)? Is there a docs page from Jupyter that covers this?

File: doc/python/webgl-vs-svg.md [hunk 2]
```diff
@@ -58,6 +59,15 @@ In addition to the above limitations, the WebGL-powered version of certain SVG-p
 * Range breaks on time-series axes are not yet supported
 * Axis range heuristics may differ
 
+### Multiple WebGL Contexts
+
+*New in 5.19*
+
+If you encounter WebGL context limits when rendering many WebGL-based figures on one page, you can use [Virtual WebGL](https://github.com/greggman/virtual-webgl).
```
Comment (alexcjohnson) [inline doc/python/webgl-vs-svg.md:66]: Can we be more precise, in most contexts I wouldn't think of 5 as "many" but that's typically the limit, right? More than 5 and it'll break without this? We can hedge and say something like "in most browsers it's possible to display up to 5 Plotly WebGL-based figures but any more and you will encounter errors"

Comment (LiamConnors) [inline doc/python/webgl-vs-svg.md:66]:
```suggestion
Most browsers have a limit of between 8 and 16 WebGL contexts per page. Each Plotly WebGL-based figure uses one context, so if you exceed your browser's limit, not all figures will render and you'll see an error. In the console in Chrome, for example, you'll see the error: "Too many active WebGL contexts. Oldest context will be lost".

If you encounter WebGL context limits when rendering WebGL-based figures on one page, you can use [Virtual WebGL](https://github.com/greggman/virtual-webgl).
```

For me I found in testing that it was 8, but I've seen other mentions of people encountering a limit at 16. And 8-16 is mentioned here https://github.com/greggman/virtual-webgl

Comment (LiamConnors) [reply doc/python/webgl-vs-svg.md]: For me I found in testing that it was 8, but I've seen other mentions of people encountering a limit at 16. And 8-16 is mentioned here https://github.com/greggman/virtual-webgl

Updated the text

File: doc/python/webgl-vs-svg.md (hunk 1 above)
Comment (LiamConnors) [reply doc/python/webgl-vs-svg.md:70]: I've added how to do it by running a cell as a HTML cell with a magic command and linked out to the docs. When testing, this was able to load the script in

File: doc/python/webgl-vs-svg.md (hunk 2 above)
Comment (alexcjohnson) [reply doc/python/webgl-vs-svg.md]: Oh you could get 8 figures? 16 is the typical number of _contexts_ a browser will allow, but in general each Plotly figure uses more than one context. I thought each one used 3 contexts (16/3 = 5.333 - ie the 6th would not have enough contexts so 5 would be the limit) but I guess either we got that down to 2 at some point, or different plot types (gl3d, gl2d, parcoords, mapbox) use different numbers of contexts.

Comment (LiamConnors) [reply doc/python/webgl-vs-svg.md]: Interesting. Yeah, I had 8 rendering in a notebook in Chrome, and it threw the warning in the console when I ran a 9th cell. I was using this example here: https://plotly.com/python/webgl-vs-svg/#webgl-with-100000-points-with-graph-objects
Does the number of contexts depend on the number of traces?

Comment (LiamConnors) [reply doc/python/webgl-vs-svg.md]: Updated the text with more details on how many figures you would likely be able to render

File: doc/python/webgl-vs-svg.md [hunk 3]
```diff
@@ -58,6 +58,25 @@ In addition to the above limitations, the WebGL-powered version of certain SVG-p
 * Range breaks on time-series axes are not yet supported
 * Axis range heuristics may differ
 
+### Multiple WebGL Contexts
+
+*New in 5.19*
+
+Most browsers have a limit of between 8 and 16 WebGL contexts per page. Plotly WebGL-based figure may use multiple WebGL contexts, but generally you'll be able to render between 4 and 8 figures on one page.
```
Comment (LiamConnors) [inline doc/python/webgl-vs-svg.md:65]:
```suggestion
Most browsers have a limit of between 8 and 16 WebGL contexts per page. A Plotly WebGL-based figure may use multiple WebGL contexts, but generally you'll be able to render between 4 and 8 figures on one page.
```

**Comment by LiamConnors** [comment]
@archmoj @alexcjohnson this is ready for another look. It now has the JS and docs updates for 2.29

**Comment by archmoj** [comment]
All looking good to me. 
Thanks very much for the PR.
:dancer:

**Review by alexcjohnson — APPROVED** [review]

**Pull request merged**